src/workerd/api/node/i18n.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#include <kj/common.h>
+#include <kj/debug.h>
+#include <kj/string.h>
+#include <kj/one-of.h>
+
+#include <unicode/putil.h>
+#include <unicode/timezone.h>
+#include <unicode/uchar.h>
+#include <unicode/uclean.h>
+#include <unicode/ucnv.h>
+#include <unicode/udata.h>
+#include <unicode/uidna.h>
+#include <unicode/ulocdata.h>
+#include <unicode/urename.h>
+#include <unicode/ustring.h>
+#include <unicode/utf16.h>
+#include <unicode/utf8.h>
+#include <unicode/utypes.h>
+#include <unicode/uvernum.h>
+#include <unicode/uversion.h>
+
+#include <string_view>
+
+namespace workerd::api::node {
+
+enum class Encoding {
+  ASCII,
+  LATIN1,
+  UTF8,
+  UTF16LE,
+  BASE64,
+  BASE64URL,
+  HEX,
+
+  // Required by transcode support.
+  UCS2,
+};
+
+Encoding getEncoding(kj::StringPtr input) {
+  if (input == "utf8") {
+    return Encoding::UTF8;
+  } else if (input == "ascii") {
+    return Encoding::ASCII;
+  } else if (input == "latin1") {
+    return Encoding::LATIN1;
+  } else if (input == "utf16le") {
+    return Encoding::UTF16LE;
+  } else if (input == "base64") {
+    return Encoding::BASE64;
+  } else if (input == "base64url") {
+    return Encoding::BASE64URL;
+  } else if (input == "hex") {
+    return Encoding::HEX;
+  }
+
+  KJ_UNREACHABLE;
+}
+
+namespace i18n {
+
+// Used by BufferUtil::transcode.
+constexpr bool canBeTranscoded(Encoding encoding) noexcept {
+  switch (encoding) {
+    case Encoding::ASCII:
+    case Encoding::LATIN1:
+    case Encoding::UCS2:
+    case Encoding::UTF8: return true;
+    default: return false;
+  }
+}
+
+
+class Converter final {
+ public:
+  explicit Converter(Encoding encoding, std::string_view substitude = "");
+  explicit Converter(UConverter* converter, std::string_view substitude = "");
+
+  UConverter* conv() const { return conv_.get(); }
+
+  size_t maxSize() const {
+    KJ_ASSERT_NONNULL(conv_.get());
+    return ucnv_getMaxCharSize(conv_.get());
+  }
+
+  size_t minSize() const {
+    KJ_ASSERT_NONNULL(conv_.get());
+    return ucnv_getMinCharSize(conv_.get());
+  }
+
+  void reset() {
+    KJ_ASSERT_NONNULL(conv_.get());
+    ucnv_reset(conv_.get());
+  }
+
+  void setSubstitudeChars(std::string_view sub) {
+    KJ_ASSERT_NONNULL(conv_.get());
+    UErrorCode status = U_ZERO_ERROR;
+    if (!sub.empty()) {
+      ucnv_setSubstChars(conv_.get(), sub.data(), sub.size(), &status);
+      KJ_ASSERT(U_SUCCESS(status));
+    }
+  }
+
+ private:
+  struct ConverterDisposer {
+    void operator()(UConverter* pointer) const { ucnv_close(pointer); }
+  };
+
+  std::unique_ptr<UConverter, ConverterDisposer> conv_;
+};
+
+using TranscodeResult = kj::OneOf<kj::Array<kj::byte>, UErrorCode>;
```

Comment:
nit: not being used?

Reply:
Oh, I see it's used in the i18n.c++ file... I would move the definition into that file if it is only used there